public/app/features/search/components/SearchResults.tsx

```diff
@@ -0,0 +1,159 @@
+import React, { FC } from 'react';
+import { css, cx } from 'emotion';
+import { GrafanaTheme } from '@grafana/data';
+import { Icon, useTheme } from '@grafana/ui';
+import { IconType } from '@grafana/ui/src/components/Icon/types';
+import { DashboardSection, ItemClickWithEvent } from '../types';
+import { SearchItem } from './SearchItem';
+import { SearchCheckbox } from './SearchCheckbox';
+
+export interface Props {
+  results: DashboardSection[] | undefined;
+  onSelectionChanged: () => void;
+  onTagSelected: (name: string) => any;
+  onFolderExpanding: () => void;
+  onToggleSelection: ItemClickWithEvent;
+  editable: boolean;
+}
+
+export const SearchResults: FC<Props> = ({
+  results,
+  onSelectionChanged,
+  onTagSelected,
+  onFolderExpanding,
+  onToggleSelection,
+  editable,
+}) => {
+  const theme = useTheme();
+  const styles = getSectionStyles(theme);
+
+  const toggleFolderExpand = (section: DashboardSection) => {
+    if (section.toggle) {
+      if (!section.expanded && onFolderExpanding) {
+        onFolderExpanding();
+      }
+
+      section.toggle(section).then(() => {
+        if (onSelectionChanged) {
+          onSelectionChanged();
+        }
+      });
+    }
+  };
+
+  // TODO display 'No results' messages after manage dashboards is refactored
+  if (!results) {
+    return null;
+  }
+
+  return (
+    <ul className={styles.wrapper}>
+      {results.map(section => (
+        <li aria-label="Search section" className={styles.section} key={section.title}>
+          <SectionHeader onSectionClick={toggleFolderExpand} {...{ onToggleSelection, editable, section }} />
+          <ul aria-label="Search items" className={styles.wrapper}>
+            {section.expanded &&
+              section.items.map(item => (
+                <SearchItem key={item.id} {...{ item, editable, onToggleSelection, onTagSelected }} />
+              ))}
+          </ul>
+        </li>
+      ))}
+    </ul>
+  );
+};
+
+const getSectionStyles = (theme: GrafanaTheme) => {
+  return {
+    wrapper: css`
+      list-style: none;
+    `,
+    section: css`
+      background: ${theme.colors.panelBg};
+      border-bottom: solid 1px ${theme.isLight ? theme.colors.gray95 : theme.colors.gray25};
+      padding: 0px 4px 4px 4px;
+      margin-bottom: 3px;
+    `,
+  };
+};
+
+interface SectionHeaderProps {
+  section: DashboardSection;
+  onSectionClick: (section: DashboardSection) => void;
+  onToggleSelection: ItemClickWithEvent;
+  editable: boolean;
+}
+
+const SectionHeader: FC<SectionHeaderProps> = ({ section, onSectionClick, onToggleSelection, editable }) => {
```

Comment:
Not sure if this should be moved to a separate file, since it's only used in `SearchResults` for now.